plugins/ufw/appprofiles.cpp

```diff
@@ -36,21 +36,17 @@ Entry::Entry(const QString &n, const QString &p)
      : name(n), ports(p)
 {
     ports.replace('|', ' ');
-    
-//     bool hasUdp=ports.contains("/udp"),
-//          hasTcp=ports.contains("/tcp);
-//          
-//     protocol=hasUdp==hasTcp ? Types::PROTO_BOTH
-//                             : hasUdp
-//                                 ? Types::PROTO_UDP
-//                                 : Types::PROTO_TCP;
 }
 
+/* TODO: Find a way to fix this.
+ this feels really wrong. a Function that returns a reference
+for a static variable created inside of it.
+*/
```

Comment:
This is the ugliest singleton I've ever seen!
I guess that proper implementation is requiered.